Automattic/wp-calypso
# Notifications: Refresh admin menu after changing comment status

Closes https://github.com/Automattic/wp-calypso/issues/94909

## Proposed Changes

* Refresh admin menu after updating comment status
  * It follows the same approach as the actions on the Comments page (`client/my-sites/comments/comment/comment-actions.jsx:87`)
  * It needed to include action handlers on the Calypso client, as the Notifications component is implemented on the apps directory and doesn't share the same store

## Why are these changes being made?

* Updating comment status doesn't update the comments count on the admin menu; it includes approving, marking as spam, and deleting

## Testing Instructions

1. Apply this PR to your local or use a test env
2. Update site setting to hold comments for moderation (Settings > Discussion > A comment is held for moderation)
3. Go to the Calypso home for the testing site (or any other page inside Calypso)
4. On an incognito tab (or another session), comment on a post from the test site
5. Open the notifications panel
6. You should see a notification for the new comment
7. Click on the notification and approve the comment
8. The comment count on the admin menu (sidebar) should update
9. Comment again and now click on Spam in the notification
10. The count on the admin menu (sidebar) should update after the undo state completes (~5 seconds later)
11. Do the same for the Trash action, and the count should also update as expected

## Pre-merge Checklist
